src/drt/src/io/io.cpp

```diff
@@ -1217,7 +1217,7 @@ void io::Parser::setRoutingLayerProperties(odb::dbTechLayer* layer,
     tmpLayer->addLef58SpacingEndOfLineConstraint(con.get());
     tech->addUConstraint(std::move(con));
   }
-  if (layer->isRectOnly()) {
+  if (layer->isRectOnly() || tech->hasUnidirectionalLayer(layer->getName())) {
     auto rectOnlyConstraint = make_unique<frLef58RectOnlyConstraint>(
         layer->isRectOnlyExceptNonCorePins());
```

Comment:
@arlpetergadfort is this what you have in mind?  This generates a rect only constraint.

Reply:
No, this will cause issues when the via enclosures are wider than the routing width.
It seems like the better place to do this would be in:
https://github.com/The-OpenROAD-Project/OpenROAD/blob/4a99e88667b0840531ca0096c4fa0da8f80d6cb1/src/drt/src/db/tech/frLayer.h#L154

Reply:
When I had to do this, I modified the number of masks, since that didn't introduce new DRCs

Reply:
I get your point. However, my only problem with editing isUnidirectional is that pin access does not check for it when generating access points which pushed me to make it rectOnly. So without rectOnly, pin access would allow for planar access in the wrong direction(if I understand correctly). @maliberty is there a good reason pin access does not use isUnidrectional instead of rectOnly? https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/drt/src/pa/FlexPA_prep.cpp#L203-L218